proto/internal.go

```diff
@@ -82,3 +83,23 @@ func (m *RaftMessageRequest) GetUser() string {
 	// TODO(marc): we should use security.NodeUser here, but we need to break cycles first.
 	return "node"
 }
+
+// NewRaftMessageRequest returns a new RaftMessageRequest.
+func NewRaftMessageRequest(groupID RangeID, msg raftpb.Message) (*RaftMessageRequest, error) {
+	m := &RaftMessageRequest{GroupID: groupID}
+	var err error
+	m.Msg, err = msg.Marshal()
+	if err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
+// UnmarshalMsg unmarshals the raft payload.
+func (m *RaftMessageRequest) UnmarshalMsg() (*raftpb.Message, error) {
+	var msg raftpb.Message
```

Comment:
this may be too early to think about, but the Raft messages seem to be good candidates to be taken from a buffer pool (to avoid alloc/GC pressure). If there are only a handful of call sites, consider unmarshaling into a `raftpb.Message` given as a parameter.